### ISSUE TITLE
Access to body of non-200 responses in stream mode

### ISSUE THREAD

**Issue by km256** [sindresorhus/got#61]
https://github.com/sindresorhus/got/blob/master/index.js#L69
Error event emits only with `err` argument and there is no way to acces to body of response.
How about somthing like 
````
cb = function () {
	var args = Array.prototype.slice.call(arguments, 0);
	args.unshift('error');
	proxy.emit.apply(proxy, args);
};
````

**Comment by floatdrop** [comment]
@km256 there is [`response`](https://github.com/sindresorhus/got#onresponse-response) event for this. Isn't it enough?

**Comment by floatdrop** [comment]
@km256 about code - it could be more verbose, PR maybe?

```js
cb = function (err, data, response) {
	proxy.emit('error', err, data, response);
};
```

**Comment by floatdrop** [comment]
Published in `3.1.0`
